src/App.tsx

```diff
@@ -1,109 +1,75 @@
-import React from 'react';
+import React, { useState } from 'react';
+
 import './App.css';
+import { Pagination } from './components/Pagination';
 import { getNumbers } from './utils';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
+const optionsPerPage = [3, 5, 10, 20];
 
 export const App: React.FC = () => {
+  const [perPage, setPerPage] = useState(optionsPerPage[1]);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const firstVisibleItemIndex = (currentPage - 1) * perPage;
+  const lastVisibleItemIndex = Math.min(42, currentPage * perPage);
```

Comment:
The use of a hardcoded value '42' makes this code less flexible. If the number of items changes, you will have to update this value as well. It's better to use 'items.length' to ensure this code adapts to the number of items automatically.